### ISSUE TITLE
Provide a non-framework code?

### ISSUE THREAD

**Issue by 18hfliu** [BioinfoMachineLearning/bio-diffusion#7]
Can you provide a non-framework code so that you can easily see the process, not just the functions? thanks! It stopped after this，and did not continue training ：
Skipping training batch with index 43 due to OOM error...
/home/ai4s-liuhaifeng/anaconda3/envs/bio-diffusion/lib/python3.9/site-packages/pytorch_lightning/loops/optimization/optimizer_loop.py:135: UserWarning: `training_step` returned `None`. If this was on 
purpose, ignore this warning...
  self.warning_cache.warn("`training_step` returned `None`. If this was on purpose, ignore this warning...")
Epoch 210/4999 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━ 44/57 0:02:03 • 0:00:35 0.38it/s loss: 3.88

**Comment by amorehead** [comment]
Hello. Are you training your model using multiple GPUs in parallel? If so, that would be the reason why your training run would stop after encountering this warning. The warning is logged here: https://github.com/BioinfoMachineLearning/bio-diffusion/blob/775a9760972f6ec254c44dd6e623c4451403ca4f/src/models/qm9_mol_gen_ddpm.py#L343. For training with a single GPU, this warning would simply skip the current training batch. However, freezing after an out-of-memory (OOM) warning is common when you have multiple GPUs running in parallel. This is because PyTorch's Distributed Data Parallel (DDP) operations may become out of sync with each GPU in the process of skipping a batch due to an OOM error. This is not an issue with the `bio-diffusion` codebase but is instead part of the broader behavior of PyTorch's DDP support.

**Comment by amorehead** [comment]
If you would like to raise and manually inspect the underlying error that your warning messages above reference, you can replace the `return` statement here (https://github.com/BioinfoMachineLearning/bio-diffusion/blob/775a9760972f6ec254c44dd6e623c4451403ca4f/src/models/qm9_mol_gen_ddpm.py#L348) with `raise e`.

**Comment by 18hfliu** [comment]
Yes, I use multiple GPUs for parallel training. Thank you very much. I have reduced the batch_size and there are no faults in 570 epochs.

**Comment by amorehead** [comment]
Great to hear!
